agent/utils/mongo_fix/mongo_fix.go

```diff
@@ -36,7 +36,16 @@ func ClientOptionsForDSN(dsn string) (*options.ClientOptions, error) { //nolint:
 	username := parsedDsn.User.Username()
 	password, _ := parsedDsn.User.Password()
 	if username != "" || password != "" {
-		clientOptions = clientOptions.SetAuth(options.Credential{Username: username, Password: password})
+		clientOptions = clientOptions.SetAuth(
+			options.Credential{
+				AuthMechanism:           clientOptions.Auth.AuthMechanism,
+				AuthMechanismProperties: clientOptions.Auth.AuthMechanismProperties,
+				AuthSource:              clientOptions.Auth.AuthSource,
+				Username:                username,
+				Password:                password,
+				PasswordSet:             clientOptions.Auth.PasswordSet,
+			},
+		)
```

Comment:
I'm not sure, but it looks like we can just do:
```
		clientOptions.Auth.Username = username
		clientOptions.Auth.Password = password
```

Reply:
yes, totally, no idea why it wasn't done initially. @artemgavrilov good catch, thank you